Xamarin.Forms.Platform.UAP/MediaElementRenderer.cs

```diff
@@ -0,0 +1,229 @@
+			}
+		}
+
+		int IMediaElementRenderer.NaturalVideoHeight
+		{
+			get	{ return Control.NaturalVideoHeight; }
+		}
+
+		int IMediaElementRenderer.NaturalVideoWidth
+		{
+			get { return Control.NaturalVideoWidth; }
+		}
+
+		TimeSpan IMediaElementRenderer.Position
+		{
+			get { return Control.Position; }
+		}
+
+		void IMediaElementRenderer.Seek(TimeSpan time)
+		{
+			Control.Position = time;
+		}
+
+		protected override void OnElementChanged(ElementChangedEventArgs<MediaElement> e)
+		{
+			base.OnElementChanged(e);
+
+			if (e.OldElement != null)
+			{
+				if (Control != null)
+				{
+					if (_positionChangedToken != 0)
+					{
+						Control.UnregisterPropertyChangedCallback(Windows.UI.Xaml.Controls.MediaElement.PositionProperty, _positionChangedToken);
+						_positionChangedToken = 0;
+					}
+
+					Control.CurrentStateChanged -= Control_CurrentStateChanged;
+					Control.SeekCompleted -= Control_SeekCompleted;
+					Control.MediaOpened -= Control_MediaOpened;
+					Control.MediaEnded -= Control_MediaEnded;
+				}
+
+				e.OldElement.SetRenderer(null);
+			}
+
+			if (e.NewElement != null)
+			{
+				SetNativeControl(new Windows.UI.Xaml.Controls.MediaElement());
+				Control.HorizontalAlignment = HorizontalAlignment.Stretch;
+				Control.VerticalAlignment = VerticalAlignment.Stretch;
+
+				e.NewElement.SetRenderer(this);
+				Control.AreTransportControlsEnabled = Element.AreTransportControlsEnabled;
+				Control.AutoPlay = Element.AutoPlay;
+				Control.IsLooping = Element.IsLooping;
+				Control.Stretch = Element.Aspect.ToStretch();
+
+				_bufferingProgressChangedToken = Control.RegisterPropertyChangedCallback(Windows.UI.Xaml.Controls.MediaElement.BufferingProgressProperty, BufferingProgressChanged);
+				_positionChangedToken = Control.RegisterPropertyChangedCallback(Windows.UI.Xaml.Controls.MediaElement.PositionProperty, PositionChanged);
+
+				Control.SeekCompleted += Control_SeekCompleted;
+				Control.CurrentStateChanged += Control_CurrentStateChanged;
+				Control.MediaOpened += Control_MediaOpened;
+				Control.MediaEnded += Control_MediaEnded;
+
+				if (Element.Source != null)
+				{
+					Control.Source = Element.Source;
+				}
+			}
+		}
+
+		void Control_MediaEnded(object sender, RoutedEventArgs e)
+		{
+			Element?.OnMediaEnded();
+		}
+
+		void Control_MediaOpened(object sender, RoutedEventArgs e)
+		{
+			Element?.RaiseMediaOpened();
+		}
+
+	
+		void Control_CurrentStateChanged(object sender, RoutedEventArgs e)
+		{
+			switch (Control.CurrentState)
+			{
+				case Windows.UI.Xaml.Media.MediaElementState.Playing:
+					if (Element.KeepScreenOn)
+					{
+						_request.RequestActive();
+					}
+					break;
+
+				case Windows.UI.Xaml.Media.MediaElementState.Paused:
+				case Windows.UI.Xaml.Media.MediaElementState.Stopped:
+				case Windows.UI.Xaml.Media.MediaElementState.Closed:
+					if (Element.KeepScreenOn)
+					{
+						_request.RequestRelease();
+					}
+					break;
+			}
+
+			if (Element != null)
```

Comment:
This does not make sense, because the above will already be NRE, in the condition `Element.KeepScreenOn`.
It is better to add the first line of the method:
```C#
if (Element == null)
return;
```